src/react/OCConnect.js

```diff
@@ -15,29 +15,48 @@ import { OCAuthLive, OCAuthSandbox } from '../sdk/auth';
 const OCConnect = ( { children, opts, sandboxMode } ) =>
 {
     const [ ocAuth, setOcAuth ] = useState( {} );
+    const [ OCId, setOCId ] = useState(null);
+    const [ ethAddress, setEthAddress ] = useState(null);
     const [ authState, setAuthState ] = useState( {
         isAuthenticated: false,
     } );
 
+    // init sdk
     useEffect( () =>
     {
         const authSdk = sandboxMode ? new OCAuthSandbox( opts ) : new OCAuthLive( opts );
         setOcAuth( authSdk );
-        updateAuthState( authSdk.getAuthState() );
     }, [] );
 
-    const updateAuthState = ( authState ) =>
+    // wait until authSdk is initialized
+    useEffect( () =>
+    {
+        if (Object.keys(ocAuth).length !== 0) {
+            updateAuthState();
+        }
+    }, [ocAuth] );
```

Comment:
don't we need to have the dependency array keyed on `ocAuth.isAuthenticated()` ?
because even internally `ocAuth` can be authenticated, it won't trigger the `userEffect` block as ocAuth itself doesn't not change. (it's the same instance). dependency checks against if the "instance" changed but not the content of the instance changed. 

Can you test this with from a fully logged out state. 
refresh
login.